### ISSUE TITLE
[share_plus] Linux and Web share URL is constructed incorrectly

### ISSUE THREAD

**Issue by stuartmorgan** [fluttercommunity/plus_plugins#235]
This line:
https://github.com/fluttercommunity/plus_plugins/blob/1dc73a16d61622836d7c0d4c6ae3a7c28923bc21/packages/share_plus_linux/lib/share_plus_linux.dart#L18
is constructing the URL incorrectly; you cannot concatenated unencoded elements together in a query string and then `encodeFull` it and get the right result, because some characters that are valid in a URL in general—notably `&`—are not valid in a query argument. If you want to construct a URL with query parameters manually, you need to encode the components explicitly.

Compare:
```
final subject = 'This & That';
final text = 'More & more';

// Incorrect:
print(Uri.encodeFull('mailto:?subject=$subject&body=$text'));
// -> mailto:?subject=This%20&%20That&body=More%20&%20more

// Correct:
print('mailto:?subject=${Uri.encodeComponent(subject)}&body=${Uri.encodeComponent(text)}');
// -> mailto:?subject=This%20%26%20That&body=More%20%26%20more
```

The first will result in an email whose subject is "That " and whose body is "More ".

**Comment by stuartmorgan** [comment]
Just noticed that this same pattern is used in the web version.

**Comment by miquelbeltran** [comment]
this should be fixed in share_plus_web 2.0.3, share_plus_linux 2.0.2 and share_plus_windows 2.0.2
